src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -68,32 +62,106 @@ private TableAttributeBindingProvider(INameResolver nameResolver, IStorageAccoun
-                bindAsyncCollector); 
+            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(
+                original.BuildJObject,
+                null,
+                original.CollectAttributeInfo);
+
+            var bindToJArray = bindingFactory.BindToExactAsyncType<TableAttribute, JArray>(
+                original.BuildJArray,
+                null,
+                original.CollectAttributeInfo);
 
             var bindingProvider = new GenericCompositeBindingProvider<TableAttribute>(
-                new IBindingProvider[] { bindToJArray, bindToJobject, bindAsyncCollector, original });
+                ValidateAttribute, nameResolver,
+                new IBindingProvider[]
+                {
+                    AllowMultipleRows(bindingFactory, bindAsyncCollector),
+                    AllowMultipleRows(bindingFactory, bindToExactCloudTable),
+                    AllowMultipleRows(bindingFactory, bindToExactTestCloudTable),
+                    bindToJArray,
+                    bindToJobject,
+                    original
+                });
 
             return bindingProvider;
         }
         
+        // Binding rule only allowed on attributes that don't specify the RowKey. 
+        private static IBindingProvider AllowMultipleRows(BindingFactory bf, IBindingProvider innerProvider)
+        {
+            return bf.AddFilter<TableAttribute>((attr, type) => attr.RowKey == null, innerProvider);
+        }
+
+        private ParameterDescriptor ToParameterDescriptorForCollector(TableAttribute attribute, ParameterInfo parameter, INameResolver nameResolver)
+        {
+            Task<IStorageAccount> t = Task.Run(() =>
+                _accountProvider.GetStorageAccountAsync(parameter, CancellationToken.None, nameResolver));
+            IStorageAccount account = t.GetAwaiter().GetResult();
+            string accountName = account.Credentials.AccountName;
+
+            return new TableParameterDescriptor
+            {
+                Name = parameter.Name,
+                AccountName = accountName,
+                TableName = Resolve(attribute.TableName),
+                Access = FileAccess.ReadWrite
+            };
+        }
+
+        private Task<IStorageTable> BindToTestCloudTable(TableAttribute attribute)
+        {
+            IStorageTable table = GetTable(attribute);
+            return Task.FromResult(table);
+        }
+
+        private async Task<CloudTable> BindToCloudTable(TableAttribute attribute)
+        {
+            IStorageTable table = GetTable(attribute);
+            await table.CreateIfNotExistsAsync(CancellationToken.None);
+
+            var sdkTable = table.SdkObject;
+            return sdkTable;
+        }
+
+        private static void ValidateAttribute(TableAttribute attribute, Type parameterType)
+        {
+            // Queue pre-existing  behavior: if there are { }in the path, then defer validation until runtime. 
```

Comment:
Do you mean Table here? What about the presence of %% characters? Can we just remove this validation and rely on the centralized runtime validation?